lib/widgets/app.dart

```diff
@@ -94,6 +94,36 @@ class ZulipApp extends StatelessWidget {
 
         shape: Border(bottom: BorderSide(color: Color(0xffcccccc))),
       ),
+      elevatedButtonTheme: ElevatedButtonThemeData(
```

Comment:
This PR covers themes for three types of buttons. Where did you look to try to get a list of the different types of button themes that are needed?

Reply:
Cool, I see.

That doc describes a migration that happened at a certain time, and I guess those were probably the only three button themes at that time. But there are more of them now.

Take a look at the documentation for the `ThemeData` class we're constructing here. If you search for "buttontheme" in that documentation (like with Ctrl+F on the [web page](https://api.flutter.dev/flutter/material/ThemeData-class.html) for that doc), you'll find there are several more properties with names like "foo button theme" for some "foo". So I think each of those is one that we'll want to set.

(This probably isn't one of the best-designed parts of Flutter's API — there should probably be a way to set this sort of thing for all of them together when desired. But for the moment we'll work with it as it is.)

Reply:
Because there's a lot of them, we'll also want to avoid duplicating the parts that are the same. (This was already true when there were just three, but it's more so when there are so many.) So please see what you can do in that direction.

Reply:
@gnprice  Sir
So I have implemented the letterSpacing for all the `button Theme`

Reply:
Please Sir
Throw my more light on this.
What exactly do we stand to achieve and at what point could there be a `duplicate part`
This would enable me know what I am looking for and also finish the task completely.

Reply:
I think Greg is saying that we want to avoid repeating a lot of the same details over and over.

When I read this PR, I see this kind of thing repeated many times:

```dart
        // Apply letter spacing to the text style for […] buttons
        style: ButtonStyle(
          textStyle: MaterialStateProperty.all<TextStyle>(
            const TextStyle(
              letterSpacing: 0.01,
            ),
          ),
        ),
```

This repetition takes many lines of code, so it makes it more difficult to find and read more interesting lines of code.

It also has a maintenance cost. For example, if our designer said he wanted 1.1% spacing instead of 1%, there would be many changes to make instead of just one, and it would be easier to accidentally miss some changes that needed to happen as part of that.

So, for example, can we extract the `ButtonStyle` value out to a new variable and reuse that variable?

Reply:
Okay 
Quite understandable
Thanks for the clarification.